File: ECommerceAPI/ECommerceAPI/Models/Menu.cs
using ECommerceAPI.Models.Interfaces;
using System;
using System.Collections.Generic;

namespace ECommerceAPI.Models
{
    public class Menu : ITreeview<Menu>
    {
        public int MenuId { get; set; }
        public int ParentId { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Url { get; set; }
        public string Image { get; set; }
        public int Sort { get; set; }
        public bool Active { get; set; }
        public int UserId { get; set; }
        public List<Role> Role { get; set; }
        public List<Menu> Children { get; set; }
        public Menu()
        {
            Children = new List<Menu>();
            Role = new List<Role>();
        }
    }
}

File: ECommerceFront/src/app/models/menu.ts
export class Menu {
    menuId: number;
    parentId: number;
    code: string;
    name: string;
    url: string;
    image: string;
    sort: number;
    active: boolean;
    expanded: boolean;
    children: Array<Menu>;

    constructor() {
        this.expanded = false;
    }
}
